Hyper/Meshes/Chunk.cs

```diff
@@ -98,36 +109,49 @@ public bool Build(Vector3 location, float deltaTime, float brushWeight, int radi
                     if (DistanceSquared(x, y, z, xi, yi, zi) <= radius * radius)
                     {
                         _voxels[xi, yi, zi].Value -= deltaTime * brushWeight * Gaussian(xi, yi, zi, x, y, z, 0.1f);
+                        flag = true;
                     }
                 }
             }
         }
 
-        UpdateMesh();
+        if(flag) UpdateMesh();
 
         var error = GL.GetError();
         if (error != ErrorCode.NoError) Logger.Error(error);
 
-        return true;
+        return flag;
     }
 
     public void UpdateCollisionSurface(Simulation simulation, BufferPool bufferPool)
     {
+        if (collisionSurface == null)
+        {
+            CreateCollisionSurface(simulation, bufferPool);
+            return;
+        }
         simulation.Shapes.RemoveAndDispose(_shape, bufferPool);
         var mesh = MeshHelper.CreateMeshFromChunk(this, bufferPool);
         _shape = simulation.Shapes.Add(mesh);
         simulation.Statics[_handle].SetShape(_shape);
     }
+    public void DisposeCollisionSurface(Simulation simulation, BufferPool bufferPool)
```

Comment:
this is never used

Reply:
usages are commented out because of an existing issue with the physics engine